src/core/swap/swap-api.ts

```diff
@@ -232,3 +249,14 @@ function rankError(error: unknown): number {
   if (asMaybeSwapCurrencyError(error) != null) return 2
   return 1
 }
+
+function isUnknownSwapError(error: unknown): boolean {
+  return (
+    asMaybeInsufficientFundsError(error) == null &&
+    asMaybePendingFundsError(error) == null &&
+    asMaybeSwapAboveLimitError(error) == null &&
+    asMaybeSwapBelowLimitError(error) == null &&
+    asMaybeSwapPermissionError(error) == null &&
+    asMaybeSwapCurrencyError(error) == null
+  )
+}
```

Comment:
Instead of building this whole new function, could we use `rankError(error) <= 1`?

Reply:
Oh, hmm.. The functions are semantically different and their is motivation for them to diverge in the future as we filter out noisy errors. Shall we opt to keep them distinct for this reason?

Reply:
If we give the error a score, then we know what it is, and by definition the error cannot be "unknown". Let's leverage this fact, but still keep the functions distinct:

```ts
function isUnknownSwapError(error: unknown): boolean {
  isKnownError = rankError(error) > 1
  return !isKnownError
}
```